More2Life/HalfModalPresentationController.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  HalfModalPresentationController.swift
+//  More2Life
+//
+//  Created by Brendan Kingsford on 8/28/17.
+//  Copyright © 2017 More2Life. All rights reserved.
+//
+
+import UIKit
+
+class HalfModalPresentationController : UIPresentationController {
+	
+    var _dimmingView: UIView?
+    var dimmingView: UIView {
+        if let dimmedView = _dimmingView {
+            return dimmedView
+        }
+        
+        let view = UIView(frame: CGRect(x: 0, y: 0, width: containerView!.bounds.width, height: containerView!.bounds.height))
+        
+        // Blur Effect
+        let blurEffect = UIBlurEffect(style: UIBlurEffectStyle.light)
+        let blurEffectView = UIVisualEffectView(effect: blurEffect)
+        blurEffectView.frame = view.bounds
+        view.addSubview(blurEffectView)
+        
+        // Vibrancy Effect
+        let vibrancyEffect = UIVibrancyEffect(blurEffect: blurEffect)
+        let vibrancyEffectView = UIVisualEffectView(effect: vibrancyEffect)
+        vibrancyEffectView.frame = view.bounds
+        
+        // Add the vibrancy view to the blur view
+        blurEffectView.contentView.addSubview(vibrancyEffectView)
+		
+		let gestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(dismissOnTap))
+		view.addGestureRecognizer(gestureRecognizer);
+		
+        _dimmingView = view
+		
+        return view
+    }
+	
+	func dismissOnTap(sender: UITapGestureRecognizer) {
+		presentingViewController.dismiss(animated: true, completion: nil)
+	}
+ 
+    override var frameOfPresentedViewInContainerView: CGRect {
+        return CGRect(x: 0, y: containerView!.bounds.height * 0.333 , width: containerView!.bounds.width, height: containerView!.bounds.height * 0.666)
```

Comment:
`!` is also known as the crash-operator. If this view doesn't have a content view then it is probably right to crash but usually I like to do something like:

```
guard let containerView = containerView else {
    fatalError("Justify the crash so everyone knows the reason why it is ok to crash.")
}
```

Reply:
Other option would be to do:

`containerView?.bounds.width ?? 0`